### PR TITLE
[CWS] improvements to jsonschema documentation extraction

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

The main goal of this PR is to allow the definition of different fields with comments instead of json struct tags.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Comment by buraizu** [comment]
Thanks, created DOCS-3640 to track as an editorial review!

File: docs/cloud-workload-security/backend.schema.json [hunk 1]
```diff
@@ -486,16 +417,15 @@
...
-        "address",
-        "offset",
-        "length",
-        "protection",
-        "flags"
+        "family",
+        "ip",
+        "port"
       ],
+      "description": "IPPortFamilySerializer is used to serialize an IP, Port and address family context to JSON"
```
Comment (kayayarai) [inline docs/cloud-workload-security/backend.schema.json:426]:
```suggestion
      "description": "IPPortFamilySerializer is used to serialize an IP, port, and address family context to JSON"
```

File: docs/cloud-workload-security/backend.md [hunk 2]
```diff
@@ -16,65 +16,1077 @@ CWS logs have the following JSON schema:
...
+                    "$ref": "#/$defs/DNSQuestion",
+                    "description": "question is a DNS question for the DNS request"
+                }
+            },
+            "additionalProperties": false,
+            "type": "object",
+            "required": [
+                "id"
+            ],
+            "description": "DNSEventSerializer serializes a dns event to JSON"
```
Comment (kayayarai) [inline docs/cloud-workload-security/backend.md:144]:
```suggestion
            "description": "DNSEventSerializer serializes a DNS event to JSON"
```
not sure where this is in the JSON, but it should be DNS

File: docs/cloud-workload-security/backend.md [hunk 3]
```diff
@@ -16,65 +16,1077 @@ CWS logs have the following JSON schema:
...
+        },
+        "NetworkContext": {
+            "properties": {
+                "device": {
+                    "$ref": "#/$defs/NetworkDevice",
+                    "description": "device is the network device on which the event was captured"
+                },
+                "l3_protocol": {
+                    "type": "string",
+                    "description": "l3_protocol is the layer 3 procotocol name"
```
Comment (kayayarai) [inline docs/cloud-workload-security/backend.md:533]:
```suggestion
                    "description": "l3_protocol is the layer 3 protocol name"
```

File: docs/cloud-workload-security/backend.md [hunk 4]
```diff
@@ -16,65 +16,1077 @@ CWS logs have the following JSON schema:
...
+                },
+                "l4_protocol": {
+                    "type": "string",
+                    "description": "l4_protocol is the layer 4 procotocol name"
```
Comment (kayayarai) [inline docs/cloud-workload-security/backend.md:537]:
```suggestion
                    "description": "l4_protocol is the layer 4 protocol name"
```

File: docs/cloud-workload-security/backend.md [hunk 5]
```diff
@@ -293,21 +1311,22 @@ CWS logs have the following JSON schema:
...
             "description": "question is a DNS question for the DNS request"
         }
     },
     "additionalProperties": false,
-    "type": "object"
+    "type": "object",
+    "required": [
+        "id"
+    ],
+    "description": "DNSEventSerializer serializes a dns event to JSON"
```
Comment (kayayarai) [inline docs/cloud-workload-security/backend.md:1329]:
```suggestion
    "description": "DNSEventSerializer serializes a DNS event to JSON"
```

File: docs/cloud-workload-security/backend.md [hunk 6]
```diff
@@ -721,7 +1744,13 @@ CWS logs have the following JSON schema:
...
     },
     "additionalProperties": false,
-    "type": "object"
+    "type": "object",
+    "required": [
+        "family",
+        "ip",
+        "port"
+    ],
+    "description": "IPPortFamilySerializer is used to serialize an IP, Port and address family context to JSON"
```
Comment (kayayarai) [inline docs/cloud-workload-security/backend.md:1753]:
```suggestion
    "description": "IPPortFamilySerializer is used to serialize an IP, port, and address family context to JSON"
```

File: docs/cloud-workload-security/backend.schema.json [hunk 7]
```diff
@@ -177,34 +106,28 @@
...
+      "type": "object",
       "required": [
-        "class",
-        "type",
-        "name",
-        "size",
-        "count"
+        "id"
       ],
+      "description": "DNSEventSerializer serializes a dns event to JSON"
```
Comment (kayayarai) [inline docs/cloud-workload-security/backend.schema.json:128]:
```suggestion
      "description": "DNSEventSerializer serializes a DNS event to JSON"
```

**Review by kayayarai — APPROVED** [review]
Some copyedit minor fixes.

File: docs/cloud-workload-security/scripts/backend-doc-gen.py [hunk 8]
```diff
@@ -46,7 +46,7 @@ def presentable_top_node(top_node):
 
 
 def extract_ref_name_and_anchor(ref):
-    prefix = "#/definitions/"
+    prefix = "#/$defs/"
```
Comment (Gui774ume) [inline docs/cloud-workload-security/scripts/backend-doc-gen.py:49]: Is it a cosmetic change, or am I missing something ?

**Review by Gui774ume — APPROVED** [review]

**Pull request merged**